src/main/kotlin/io/github/gradlenexus/publishplugin/internal/InvalidatingStagingRepositoryDescriptorRegistry.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.gradlenexus.publishplugin.internal
+
+import org.gradle.api.artifacts.repositories.ArtifactRepository
+import org.gradle.api.internal.artifacts.repositories.AbstractResolutionAwareArtifactRepository
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.util.concurrent.ConcurrentHashMap
+
+/*
+ * This class is a temporary workaround to invalidate the repository description on a IvyArtifactRepository after the creation of the staging sonatype repo
+ */
+class InvalidatingStagingRepositoryDescriptorRegistry : StagingRepositoryDescriptorRegistry() {
+
+    private val mapping = ConcurrentHashMap<String, StagingRepositoryDescriptor>()
```

Comment:
Do we need `mapping` here? `toString()` which seems to be the only to use it, could be reused from the super class.

Reply:
no, not at all